libs/store/src/lib/store/utils/dispatch-data-to-store/dispatch-data-to-store.util.ts

```diff
@@ -1,37 +1,159 @@
  * @param data - The data we wish to dispatch to the store
  * @param store - The store we wish to dispatch the data to
  * @param dispatchType - Whether we wish to set, add or update the data. By default this is `set`
- * @return {*}  {Observable<DataType>}
+ * @returns {Observable<(DataType | DataType[] | null)>} - Returns the data that was dispatched to the store. The generic type is the same as the `data`.
  */
-export const dispatchDataToStore = <DataType>(
-	dispatchAction: any,
-	data: Observable<DataType>,
+export function dispatchDataToStore<DataType = any>(
+	dispatchAction: BaseStoreActions<DataType> | EntityStoreActions<DataType>,
+	selectAction: BaseStoreSelectors<DataType> | EntityStoreSelectors<DataType> | undefined,
+	data: Observable<DataType | DataType[]>,
 	store: Store,
 	dispatchType: 'set' | 'add' | 'update' = 'set'
-): Observable<DataType> => {
+): Observable<DataType | DataType[]> | null {
 	// Iben: Set the loading state to true and the error state to false to start a new set
 	store.dispatch(dispatchAction.loading({ payload: true }));
 	store.dispatch(dispatchAction.error({ payload: false }));
 
 	// Iben: Set, add or update the data according to the provided dispatch type
 	return data.pipe(
 		tap((payload) => {
-			if (dispatchType === 'set') {
-				store.dispatch(dispatchAction.set({ payload }));
-			} else if (dispatchType === 'add') {
-				store.dispatch(dispatchAction.add({ payload }));
-			} else {
-				store.dispatch(dispatchAction.update({ payload }));
+			switch (dispatchType) {
+				case 'set': {
+					return payload instanceof Array
+						? store.dispatch(
+								(dispatchAction as EntityStoreActions<DataType>).set({
+									payload,
+								})
+						  )
+						: store.dispatch(
+								(dispatchAction as BaseStoreActions<DataType>).set({
+									payload,
+								})
+						  );
+				}
+				case 'add': {
+					if (!(payload instanceof Array)) {
+						return throwError(
+							() =>
+								new Error(
+									'NgxStore: Payload must be an array when using "add". Also make sure your storeslice is of type "EntityStoreAssets".'
+								)
+						);
+					}
+
+					return store.dispatch(
+						(dispatchAction as EntityStoreActions<DataType>).add({ payload })
+					);
+				}
+				case 'update': {
+					if (payload instanceof Array) {
+						return throwError(
+							() =>
+								new Error(
+									'NgxStore: Payload must not be an array when using "update".'
+								)
+						);
+					}
+
+					return store.dispatch(
+						(dispatchAction as EntityStoreActions<DataType>).update({
+							payload,
+						})
+					);
+				}
+			}
+		}),
+		switchMap((payload) => {
+			if (!selectAction) {
+				return of(null);
 			}
+
+			// Wouter: Get the value recently set to the store. This is done to keep the Store as SSOT.
+			return store.select(
+				payload instanceof Array
```

Comment:
I think it's better to run a check on the selectAction rather than on the payload, because again, that's not possible with how the store works.

You'll be better off doing something like `selectAction['selectAll'] ? selectAction.selectAll : selectAction.select`